src/cargo/sources/git/utils.rs

```diff
@@ -694,12 +695,36 @@ pub fn with_fetch_options(
     let mut progress = Progress::new("Fetch", config);
     network::with_retry(config, || {
         with_authentication(url, git_config, |f| {
+            let mut last_update = Instant::now();
             let mut rcb = git2::RemoteCallbacks::new();
+            let mut counter = MetricsCounter::<10>::new(0);
             rcb.credentials(f);
-
             rcb.transfer_progress(|stats| {
+                let indexed_deltas = stats.indexed_deltas();
+                let msg = if indexed_deltas > 0 {
+                    // Resolving deltas.
+                    format!(
+                        ", ({}/{}) resolving deltas",
+                        indexed_deltas,
+                        stats.total_deltas()
+                    )
+                } else {
+                    // Receiving objects.
+                    let duration = last_update.elapsed();
+                    if duration > Duration::from_millis(300) {
```

Comment:
Could this have a comment for why 300 and 10 above are chosen? (aka commenting 300 * 10 = 3s).

Also, the call to `elapsed()` here calls `Instant::now`  but it's also called again in `counter.add`, so could this pass `Instant::now` as a parameter to `counter`?

Reply:
👍🏾 
Fixed. The `counter::new` and related unit tests also updated.